pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTAnnotation.java

```diff
@@ -4,20 +4,53 @@
 
 package net.sourceforge.pmd.lang.apex.ast;
 
-import com.google.summit.ast.Node;
 import java.util.Arrays;
 import java.util.Set;
 import java.util.TreeSet;
 
 import net.sourceforge.pmd.Rule;
 import net.sourceforge.pmd.annotation.InternalApi;
 
-public class ASTAnnotation extends AbstractApexNode.Single<Node> {
+import com.google.common.collect.ImmutableSet;
+import com.google.summit.ast.modifier.AnnotationModifier;
+
+public class ASTAnnotation extends AbstractApexNode.Single<AnnotationModifier> {
+
+    private static final ImmutableSet<String> VALID_ANNOTATION_NAMES = ImmutableSet.of(
+            "allowcertifiedapex",
```

Comment:
Did these come from Summit or somewhere on the web?  Would be good to add a comment with a link if possible.

Reply:
They were extracted from a decompiled Jorje class, but most (not all) can be found on this [official page](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation.htm). The missing annotations are `AllowCertifiedApex`, `HiddenFromDoc`, `NamespaceGuard`, `PermGuard`, `PrivateApi`, `UseConnectDeserializer`, `UseConnectSerializer`, and `VisibleApiVersion`, which are also recognized by Jorje. How do you recommend this to be documented in this case?

Reply:
I would put exactly that in the comment (link to official page + a callout about the additional cases that exist for backward compatibility).